### ISSUE TITLE
Pygame still a dependency, fatal error if it is not installed?

### ISSUE THREAD

**Issue by will-ca** [kivy/kivy#5872]
### Versions

* Python: 
```
$ python --version
Python 2.7.15

$ python3 --version
Python 3.6.5
```
* OS:
```
Linux 4.17.2-100.fc27.x86_64 SMP x86_64 GNU/Linux
```
* Kivy:
```
$ pip show kivy
Name: Kivy
Version: 1.10.1
Summary: A software library for rapid development of hardware-accelerated multitouch applications.
Home-page: http://kivy.org
Author: Kivy Team and other contributors
Author-email: kivy-dev@googlegroups.com
License: MIT
Location: /home/Username/.local/lib/python2.7/site-packages
Requires: Kivy-Garden, docutils, pygments

$ pip3 show kivy
Name: Kivy
Version: 1.10.1
Summary: A software library for rapid development of hardware-accelerated multitouch applications.
Home-page: http://kivy.org
Author: Kivy Team and other contributors
Author-email: kivy-dev@googlegroups.com
License: MIT
Location: /home/Username/.local/lib/python3.6/site-packages
Requires: Kivy-Garden, docutils, pygments
```
* Kivy installation method:
```
$ pip install --user kivy
$ pip3 install --user kivy
$ dnf install make mercurial automake gcc gcc-c++ SDL_ttf-devel SDL_mixer-devel SDL2_ttf-devel SDL2_mixer-devel khrplatform-devel mesa-libGLES mesa-libGLES-devel gstreamer-plugins-good SDL2_image-devel SDL2-devel gstreamer gstreamer-python gstreamer1-devel pygame mtdev-devel python-devel python-pip python3-devel python3-pip
```
### Description

I am trying to get the basic `Hello World` example at the bottom of the `kivy.org` home page to run and display the window with a single button. It works fine as expected using Python2, but aborts with a fatal error in Python3.

After using `pip3` to install `pygame` for Python3, it works fine in Python3 as well. However, there is [commentary](https://github.com/kivy/kivy/issues/4085#issuecomment-323504048) that Kivy should not require `pygame` as a dependency anymore as it is not [recommended](https://github.com/kivy/kivy/issues/5862#issuecomment-405364618), `pygame` is referenced in the Linux [installation instructions](https://kivy.org/docs/installation/installation-linux.html) only as a "legacy" requirement, and if Kivy is only resorting to `pygame` on the first run as some sort of fallback due to the absence of some other required package, then there is insufficient information printed to indicate this to a first-time user.

### Code and Logs
```
$ echo "from kivy.app import App
from kivy.uix.button import Button

class TestApp(App):
    def build(self):
        return Button(text='Hello World')

TestApp().run()" > Example.py
```
```
$ python Example.py
[INFO   ] [Logger      ] Record log in /home/Username/.kivy/logs/kivy_18-07-24_4.txt
[INFO   ] [Kivy        ] v1.10.1
[INFO   ] [Python      ] v2.7.15 (default, May 15 2018, 15:37:31) 
[GCC 7.3.1 20180303 (Red Hat 7.3.1-5)]
[INFO   ] [Factory     ] 194 symbols loaded
[INFO   ] [Image       ] Providers: img_tex, img_dds, img_pygame, img_pil, img_gif (img_ffpyplayer ignored)
[INFO   ] [Text        ] Provider: pygame
[INFO   ] [Window      ] Provider: pygame(['window_egl_rpi'] ignored)
[INFO   ] [GL          ] Using the "OpenGL" graphics system
[INFO   ] [GL          ] Backend used <gl>
[INFO   ] [GL          ] OpenGL version <4.6.0 NVIDIA 390.59>
[INFO   ] [GL          ] OpenGL vendor <NVIDIA Corporation>
[INFO   ] [GL          ] OpenGL renderer <GeForce GTX 760/PCIe/SSE2>
[INFO   ] [GL          ] OpenGL parsed version: 4, 6
[INFO   ] [GL          ] Shading version <4.60 NVIDIA>
[INFO   ] [GL          ] Texture max size <16384>
[INFO   ] [GL          ] Texture max units <32>
[INFO   ] [Window      ] virtual keyboard not allowed, single mode, not docked
[INFO   ] [Base        ] Start application main loop
[INFO   ] [GL          ] NPOT texture support is available
[INFO   ] [Base        ] Leaving application in progress...
```
*Works as expected.*
![Yes, I just took the image straight from the homepage.](https://kivy.org/images/helloworld.png)
```
$ python3 Example.py
[INFO   ] [Logger      ] Record log in /home/Username/.kivy/logs/kivy_18-07-24_5.txt
[INFO   ] [Kivy        ] v1.10.1
[INFO   ] [Python      ] v3.6.5 (default, Apr  4 2018, 15:01:18) 
[GCC 7.3.1 20180303 (Red Hat 7.3.1-5)]
[INFO   ] [Factory     ] 194 symbols loaded
[INFO   ] [Image       ] Providers: img_tex, img_dds, img_pil, img_gif (img_pygame, img_ffpyplayer ignored)
[INFO   ] [Text        ] Provider: pil(['text_pygame'] ignored)
[CRITICAL] [Window      ] Unable to find any valuable Window provider.
egl_rpi - ImportError: cannot import name 'bcm'
  File "/home/Username/.local/lib/python3.6/site-packages/kivy/core/__init__.py", line 59, in core_select_lib
    fromlist=[modulename], level=0)
  File "/home/Username/.local/lib/python3.6/site-packages/kivy/core/window/window_egl_rpi.py", line 12, in <module>
    from kivy.lib.vidcore_lite import bcm, egl

pygame - ModuleNotFoundError: No module named 'pygame'
  File "/home/Username/.local/lib/python3.6/site-packages/kivy/core/__init__.py", line 59, in core_select_lib
    fromlist=[modulename], level=0)
  File "/home/Username/.local/lib/python3.6/site-packages/kivy/core/window/window_pygame.py", line 8, in <module>
    import pygame

x11 - ModuleNotFoundError: No module named 'kivy.core.window.window_x11'
  File "/home/Username/.local/lib/python3.6/site-packages/kivy/core/__init__.py", line 59, in core_select_lib
    fromlist=[modulename], level=0)

[CRITICAL] [App         ] Unable to get a Window, abort.
```
*Does not work.*
```
$ pip3 install --user pygame
Collecting pygame
  Downloading https://files.pythonhosted.org/packages/b3/5e/fb7c85304ad1fd52008fd25fce97a7f59e6147ae97378afc86cf0f5d9146/pygame-1.9.4-cp36-cp36m-manylinux1_x86_64.whl (12.1MB)
    100% |████████████████████████████████| 12.1MB 113kB/s 
Installing collected packages: pygame
Successfully installed pygame-1.9.4
$ python3 Example.py
[INFO   ] [Logger      ] Record log in /home/Username/.kivy/logs/kivy_18-07-24_7.txt
[INFO   ] [Kivy        ] v1.10.1
[INFO   ] [Python      ] v3.6.5 (default, Apr  4 2018, 15:01:18) 
[GCC 7.3.1 20180303 (Red Hat 7.3.1-5)]
[INFO   ] [Factory     ] 194 symbols loaded
pygame 1.9.4
Hello from the pygame community. https://www.pygame.org/contribute.html
[INFO   ] [Image       ] Providers: img_tex, img_dds, img_pygame, img_pil, img_gif (img_ffpyplayer ignored)
[INFO   ] [Text        ] Provider: pygame
[INFO   ] [Window      ] Provider: pygame(['window_egl_rpi'] ignored)
[INFO   ] [GL          ] Using the "OpenGL" graphics system
[INFO   ] [GL          ] Backend used <gl>
[INFO   ] [GL          ] OpenGL version <b'4.6.0 NVIDIA 390.59'>
[INFO   ] [GL          ] OpenGL vendor <b'NVIDIA Corporation'>
[INFO   ] [GL          ] OpenGL renderer <b'GeForce GTX 760/PCIe/SSE2'>
[INFO   ] [GL          ] OpenGL parsed version: 4, 6
[INFO   ] [GL          ] Shading version <b'4.60 NVIDIA'>
[INFO   ] [GL          ] Texture max size <16384>
[INFO   ] [GL          ] Texture max units <32>
[INFO   ] [Window      ] virtual keyboard not allowed, single mode, not docked
[INFO   ] [Base        ] Start application main loop
[INFO   ] [GL          ] NPOT texture support is available
[INFO   ] [Base        ] Leaving application in progress...
```
*Works as expected.*

**Comment by dolang** [comment]
Yes, pygame is legacy and should not be used anymore, although it still works.

To get a correct build with SDL2 you need a few dependencies. The most important ones are the 4 SDL2 dependencies, but it looks like you should have those. I haven't tried it on Fedora 27, but I've made an Ansible playbook for Fedora 28 and [here are its dependencies](https://gist.github.com/dolang/6b58c49836b525ad8fb94618e260ac4b#file-kivy-dev-fedora-yml-L24-L30) (you don't need to be familiar with Ansible, files are quite readable).

The docs are a bit outdated, it's something I'm working on.

Can uninstall, check your dependencies and try again? If you run `pip install` with `-v` you'll see the compilation output. If it still fails, please post that here, too.

**Comment by will-ca** [comment]
Thanks a bunch for the quick reply. I followed your instructions to reinstall everything and it works fine now without `pygame`.

I think the mistake I made last time was that I actually installed a couple of the dependencies only after I'd already installed Kivy. So then, I assume, by that point the build with SDL2 had already failed and the only way I could get it to work was to install `pygame` which it could then fall back to at run time.
